compiler/rustc_typeck/src/check/place_op.rs

```diff
@@ -82,7 +82,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             expr, base_expr, adjusted_ty, index_ty
         );
 
-        for &unsize in &[false, true] {
+        for unsize in [false, true] {
```

Comment:
Not something that needs changing in this PR: I wonder, in general, how perf differs between copying an array vs slice-iterating the rvalue-static-promoted one.

Reply:
See the perf run, looks like not much: https://github.com/rust-lang/rust/pull/86311#issuecomment-861079643. But it might just be because these aren't in perf-sensitive parts of the compiler.